terror/terror.go

```diff
@@ -106,6 +118,55 @@ type Error struct {
 }
 
 // Error implements error interface.
-func (te *Error) Error() string {
-	return fmt.Sprintf("[%s:%d]%s", te.Class, te.Code, te.Message)
+func (e *Error) Error() string {
+	return fmt.Sprintf("[%s:%d]%s", e.Class, e.Code, e.Message)
+}
+
+// Gen generates a new *Error with the same class and code, and a new formatted message.
+func (e *Error) Gen(format string, args ...interface{}) *Error {
```

Comment:
New is usually used to name a function rather than method.

And ‘New' doesn't have the implication that the created instance has the same class and error code.